src/hotspot/share/opto/intrinsicnode.hpp

```diff
@@ -198,6 +198,7 @@ class EncodeISOArrayNode: public Node {
   virtual uint ideal_reg() const { return Op_RegI; }
   virtual Node* Ideal(PhaseGVN* phase, bool can_reshape);
   virtual const Type* Value(PhaseGVN* phase) const;
+  virtual uint size_of() const { return sizeof(EncodeISOArrayNode); }
```

Comment:
Shouldn't nodes that define fields also override `Node::cmp` and `Node::hash`?

Reply:
I checked some nodes. `cmp/hash` are not always updated for new fields. It looks "nice to have". I have added `cmp/hash` for “EncodeISOArrayNode/ClearArrayNode/OpaqueMultiversioningNode" .

Reply:
Thanks for doing that but it's not only nice to have, right? GVN might otherwise incorrectly common two different nodes.

Reply:
I found `OpaqueNode` can not have `cmp` and `hash`, it will fail some tests. I think these nodes can not be optimized by GVN.